pkg/provisioner/ironic/clients/auth.go

```diff
@@ -0,0 +1,95 @@
+package clients
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"path"
+	"path/filepath"
+)
+
+// AuthType is the method of authenticating requests to the server
+type AuthType string
+
+const (
+	// NoAuth uses no authentication
+	NoAuth AuthType = "noauth"
+	// HTTPBasicAuth uses HTTP Basic Authentication
+	HTTPBasicAuth AuthType = "http_basic"
+)
+
+// AuthConfig contains data needed to configure authentication in the client
+type AuthConfig struct {
+	Type     AuthType
+	Username string
+	Password string
+}
+
+func (auth *AuthConfig) load(clientType string) error {
```

Comment:
Making this a method of AuthConfig instead of a factory feels odd. This function could return an error and the user would have a partially configured AuthConfig instance that they could try to use.

Reply:
It's private, but it is a bit weird. I'll change it.